src/rprblender/export/image.py

```diff
@@ -92,84 +90,58 @@ def sync(rpr_context, image: bpy.types.Image):
-
-    def is_empty(self) -> bool:
-        return self.pixels is None
-
-    def extract_pixels(self, image: bpy.types.Image, x1, y1, x2, y2) -> np.array:
-        """ Store source image pixels cropped to region coordinates """
-        # extract pixels
-        data = np.fromiter(image.pixels, dtype=np.float32,
-                           count=image.size[0] * image.size[1] * image.channels)
-        pixels = data.reshape(image.size[1], image.size[0], image.channels)
-
-        return self.extract_pixels_region(pixels, x1, y1, x2, y2)
+            raise ValueError("Image has no data", image)
 
-    def extract_pixels_region(self, pixels: np.array, x1, y1, x2, y2) -> np.array:
-        """ Crop pixels to region size """
-        region_pixels = np.array(pixels[y1:y2+1, x1:x2+1, :], dtype=np.float32)
+        pixels = image.pixels
+        if hasattr(pixels, 'foreach_get'):
+            data = utils.get_prop_array_data(pixels)
+        else:
+            # loading image by pixels
+            data = np.fromiter(pixels, dtype=np.float32,
+                               count=image.size[0] * image.size[1] * image.channels)
 
-        return region_pixels
+        self.pixels = data.reshape(image.size[1], image.size[0], image.channels)
 
-    def export_full(self, rpr_context) -> (pyrpr.Image, None):
-        """ Export the full image pixels as RPR image"""
-        if self.is_empty():
-            return None
+        self.name = image.name
+        self.color_space = image.colorspace_settings.name
 
-        image_key = f"{self.name}@{self.color_space}"
+    def export(self, rpr_context, render_size=None, tile=((0, 0), (1, 1))):
+        """Export pixels cropped to render and tile size as RPR image"""
 
-        if image_key in rpr_context.images:
-            return rpr_context.images[image_key]
+        image_size = self.pixels.shape[1], self.pixels.shape[0]
+        if render_size:
+            image_ratio = image_size[0] / image_size[1]
+            render_ratio = render_size[0] / render_size[1]
 
-        pixels = np.ascontiguousarray(np.flipud(self.pixels))
-        rpr_image = rpr_context.create_image_data(image_key, pixels)
-        rpr_image.set_name(image_key)
+            if image_ratio > render_ratio:
+                size = image_size[1] * render_ratio, image_size[1]
+            else:
+                size = image_size[0], image_size[0] / render_ratio
 
-        if self.color_space in ('sRGB', 'BD16', 'Filmic Log'):
-            rpr_image.set_gamma(2.2)
+            x1, y1 = (image_size[0] - size[0]) / 2, (image_size[1] - size[1]) / 2
+            x2, y2 = x1 + size[0], y1 + size[1]
 
-        return rpr_image
+        else:
+            x1, y1 = 0, 0
+            x2, y2 = image_size
 
-    def export_region(self, rpr_context, x1, y1, x2, y2) -> (pyrpr.Image, None):
-        """ Export pixels cropped to sub-region coordinates as RPR image """
-        if self.is_empty():
-            return None
+        x1, y1, x2, y2 = (
```

Comment:
Just a comment, not really sure what assigning these in a tuple fashion gains, but looks fine ;)

Reply:
I'm assigning here image rectangle coordinates due to tile values.